examples/Utils/Feeds.php

```diff
@@ -0,0 +1,178 @@
+        // Constructs the query to get the feed item with attribute values.
+        $query = "SELECT feed_item.attribute_values FROM feed_item"
+            . " WHERE feed_item.resource_name = '$feedItemResourceName'";
+        // Issues a search request by specifying page size.
+        $response =
+            $googleAdsServiceClient->search($customerId, $query, ['pageSize' => self::PAGE_SIZE]);
+
+        // Returns the feed item attribute values, which belongs to the first item. We can ensure
+        // it belongs to the first one because we specified the feed item resource name in the
+        // query.
+        return $response->getIterator()->current()->getFeedItem();
+    }
+
+    /**
+     * Gets the index of the target feed item attribute value. This is needed to specify which feed
+     * item attribute value will be updated in the given feed item.
+     *
+     * @param FeedItemAttributeValue $targetFeedItemAttributeValue the new feed item attribute value
+     *     that will be updated
+     * @param FeedItem $feedItem the feed item that will be updated
+     */
+    public static function attributeIndexFor(
+        FeedItemAttributeValue $targetFeedItemAttributeValue,
+        FeedItem $feedItem
+    ) {
+        $attributeIndex = -1;
+        // Loops through attribute values to find the index of the feed item attribute value to
+        // update.
+        foreach ($feedItem->getAttributeValues() as $feedItemAttributeValue) {
+            /** @var FeedItemAttributeValue $feedItemAttributeValue */
+            $attributeIndex++;
+            // Checks if the current feedItemAttributeValue is the one we are updating
+            if ($feedItemAttributeValue->getFeedAttributeIdUnwrapped()
+                === $targetFeedItemAttributeValue->getFeedAttributeIdUnwrapped()) {
+                break;
+            }
+        }
+
+        if ($attributeIndex === -1) {
+            throw new \InvalidArgumentException(
+                'No matching feed attribute for feed item attribute ID: '
+                . $targetFeedItemAttributeValue->getFeedAttributeIdUnwrapped()
+            );
+        }
+
+        return $attributeIndex;
+    }
+
+    /**
+     * Retrieves the place holder fields to field attributes map for a flights feed.
+     *
+     * @see Feeds::placeholderFieldsMapFor()
+     *
+     * @param string $feedResourceName the feed resource name to get the attributes from
+     * @param int $customerId the customer ID
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @return array the map from placeholder fields to feed attributes
+     */
+    public static function flightPlaceholderFieldsMapFor(
+        string $feedResourceName,
+        int $customerId,
+        GoogleAdsClient $googleAdsClient
+    ) {
+        return self::placeholderFieldsMapFor(
+            $feedResourceName,
+            $customerId,
+            $googleAdsClient,
+            function (FeedAttribute $feedAttribute) {
```

Comment:
Why not providing a map here instead of mapping function as a parameter?

The value here would be something like:
[
'Flight Description' => FlightPlaceholderField::FLIGHT_DESCRIPTION,
'Destination ID' => FlightPlaceholderField::DESTINATION_ID,
...
]

And a generic function that leverages this mapping parameter would be implemented for all in the 'placeholderFieldsMapFor' method

Reply:
That's because I need to use array_map which requires a callable function.
It doesn't require an associative array.
Or do you have other ways around to propose for the generic function too?

Reply:
I think you could make it work with the `use` keyword to introduce the associative array as a local scope variable of the callable function and rework this last to be generic. Something like that:

```php
$placeHolderFieldMappings = [
    'Flight Description' => FlightPlaceholderField::FLIGHT_DESCRIPTION,
    'Destination ID' => FlightPlaceholderField::DESTINATION_ID,
    ...
]

$callable = function (FeedAttribute $feedAttribute) use ($placeHolderFieldMappings) {
    // TODO: return RuntimeException if the given feed attribute is undefined in the mappings
    return $placeHolderFieldMappings[$feedAttribute];
}

return self::placeholderFieldsMapFor(
            $feedResourceName,
            $customerId,
            $googleAdsClient,
            $callable
);
```

WDYT?

Reply:
That looks great to me too so I just changed to it.